### PR TITLE
runtime: Improve `Map` instrumentation

### PR DESCRIPTION
**Pull request by fmeum**

### CODE REVIEW

**Comment by fmeum** [comment]
Stacked on #843

File: src/main/java/com/code_intelligence/jazzer/runtime/TraceCmpHooks.java [hunk 1]
```diff
@@ -546,31 +546,56 @@ private static void guideTowardContainmentOfFirstElement(
...
+  }
+
+  @MethodHook(type = HookType.AFTER, targetClassName = "java.util.Map",
+      targetMethod = "getOrDefault",
+      targetMethodDescriptor = "(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;")
+  public static void
+  mapGetOrDefault(
+      MethodHandle method, Object thisObject, Object[] arguments, int hookId, Object value) {
+    Object defaultValue = arguments[1];
+    if (Objects.equals(value, defaultValue)) {
```
[the remark this answers is not in the document]
Comment (fmeum) [reply src/main/java/com/code_intelligence/jazzer/runtime/TraceCmpHooks.java]: Nice, done.

**Comment by fmeum** [comment]
> As we're at it, should we also hook `remove` and `replace` to potentially mutate maps in another interesting ways?

I thought about that but couldn't come up with common cases in which this would meaningfully influence control flow. For example, most of the time the return value of `remove` isn't used and the function is called on something known to be contained in the map. I'm undecided.

**Comment by bertschneider** [comment]
Then let's think about this a little longer. Perhaps we find a proper use-case.

**Comment by fmeum** [comment]
> Then let's think about this a little longer. Perhaps we find a proper use-case.

Just to understand your point better: Would you prefer to think about this more *after* or *before* we consider merging this PR? I'm fine with both :-)

**Comment by bertschneider** [comment]
Ah, afterwards! :smile:

**Comment by fmeum** [comment]
> Ah, afterwards! Let's get this merged and perhaps our subconscious comes up with a valid use-case 😄

*hits auto-merge* :-)

**Pull request merged**